types/ramda/test/converge-tests.ts

```diff
@@ -5,101 +5,92 @@ import * as R from 'ramda';
+    ] as const);
 
-() => {
-    function add(a: number, b: number) {
-        return a + b;
-    }
+    // $ExpectType string
+    const indented = format({ indent: 2, value: 'foo\nbar\nbaz\n' }); // => '  foo\n  bar\n  baz\n'
 
-    function multiply(a: number, b: number) {
-        return a * b;
-    }
+    const add = (a: number, b: number) => a + b;
+    const multiply = (a: number, b: number) => a * b;
+    const subtract = (a: number, b: number) => a - b;
+    const concat = (a: string, b: string) => a + b;
 
-    function subtract(a: number, b: number) {
-        return a - b;
-    }
+    const add3 = (a: number, b: number, c: number) => a + b + c;
+    const add10 = (...args: [number, number, number, number, number, number, number, number, number, number]) =>
+        R.sum(args);
 
-    function concat(a: string, b: string) {
-        return a + b;
-    }
+    // $Expect Curry<() => void>
+    const emptyFunction = R.converge(() => {}, []);
 
-    // ≅ multiply( add(1, 2), subtract(1, 2) );
-    // $ExpectType (a: number, b: number) => number
+    // fn(1, 2) ≅ multiply( add(1, 2), subtract(1, 2) );
+    // $ExpectType Curry<(a: number, b: number) => number>
     const fn = R.converge(multiply, [add, subtract]);
 
     // $ExpectError
-    R.converge(concat, [add, subtract]);
+    const fnMismatchedTypes = R.converge(concat, [add, subtract]);
 
     // $ExpectError
-    R.converge(multiply, [add, subtract, add]);
+    const fnWrongNumberOfBranchesV1 = R.converge(concat, []);
 
     // $ExpectError
-    R.converge(concat, []);
+    const fnWrongNumberOfBranchesV2 = R.converge(add10, [add, add, add, add, add, add, add, add, add, add, add, add]);
+
+    // because fifth function in branches has arity of 3 result function also must have largest arity
+    // $ExpectType Curry<(a: number, b: number, c: number) => number>
+    const fn10 = R.converge(add10, [add, add, add, add, add3, add, add, add, add, add]);
+
+    const args1 = (a1: number | string) => 1;
+    const args2 = (a1: number | bigint, a2: { q: string }) => 2;
+    const args3 = (a1: number | symbol, a2: { w: number }, a3: number) => 3;
 
+    // resulted function must accept types of parameters that will satisfy every function in branches, so intersection must be used
+    // $ExpectType Curry<(a1: number, a2: { w: number; } & { q: string; }, a3: number) => number>
+    const intersectionOfArguments = R.converge(add3, [args1, args2, args3]);
+
+    const argsIncompatible = (a1: string) => 1;
+
+    // types: `string`, `number | symbol`, and `number | bigint` - do not have common type
     // $ExpectError
-    R.converge(() => {}, []);
+    const intersectionOfArgumentsIncompatible = R.converge(add3, [argsIncompatible, args2, args3]);
 
     // $ExpectType number
-    fn(1, 2);
+    const resultNumber = intersectionOfArguments(1, { q: 'text', w: 22 }, 11);
 
     // $ExpectError
-    fn('1', 2);
+    const errorArguments = intersectionOfArguments(1, { q: 'text' }, 11);
```

Comment:
Why was the original test removed? (see review comment)

Reply:
And actually all tests are the same(in context of logic). I just added variables and tried to make them more readable.
But anyway, I added back the previous ones.

Reply:
If you only want to make test more readable I would do so in a separate commit. So it's easy to check if this rewriting didn't rely on changing the implementation. I guess it's a stylistic choice but it makes review easier if you reduce the number of things changed at once.